File: Problem02/Problem02.py
file = open("input.txt", "r")
input = file.readlines()

def part1(input):
    horizontal = 0;
    depth = 0;
    for i in range(0, len(input)):
        data = input[i].split()
        if (data[0] == "forward"):
            horizontal += int(data[1])
        elif (data[0] == "up"):
            depth -= int(data[1])
        elif (data[0] == "down"):
            depth += int(data[1])
    return (horizontal * depth)

def part2(input):
    horizontal = 0;
    depth = 0;
    aim = 0;
    for i in range(0, len(input)):
        data = input[i].split()
        if (data[0] == "forward"):
            horizontal += int(data[1])
            depth += (int(data[1]) * aim)
        elif (data[0] == "up"):
            aim -= int(data[1])
        elif (data[0] == "down"):
            aim += int(data[1])
    return (horizontal * depth)

print("Part 1: " + str(part1(input)) )
print("Part 2: " + str(part2(input)) )

File: Problem02/Problem02.java
import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

public class Problem02 {
  //================================================================================
  // TESTER
  //================================================================================
  public static void main(String[] args) {
    int horizontal = 0;
    int depth = 0;
    int aim = 0;

    try {
      String line;
      File file = new File(args[0]);
      Scanner input = new Scanner(file);
      while (input.hasNextLine()) {
        line = input.nextLine();
        String[] data = line.split(" ");

        try {

          String mode = args[1];
          if (mode.equals("1")) {
            if (data[0].equals("forward")) {
              horizontal += Integer.parseInt(data[1]);
            } else if (data[0].equals("up")) {
              depth -= Integer.parseInt(data[1]);
            } else if (data[0].equals("down")) {
              depth += Integer.parseInt(data[1]);
            }
          } else if (mode.equals("2")) {
            if (data[0].equals("forward")) {
              horizontal += Integer.parseInt(data[1]);
              depth += Integer.parseInt(data[1]) * aim;
            } else if (data[0].equals("up")) {
              aim -= Integer.parseInt(data[1]);
            } else if (data[0].equals("down")) {
              aim += Integer.parseInt(data[1]);
            }
          }

        } catch (ArrayIndexOutOfBoundsException e) {
          System.out.println("Invalid syntax: use \"java Problem02 [filename] [1/2]\".");
        }
      }
      input.close();
    } catch (FileNotFoundException ex) {
      System.out.println("File not found!");
    }

    System.out.println(horizontal*depth);
  }
}
